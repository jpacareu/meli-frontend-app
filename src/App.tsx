import React from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import './styles/main.scss';
import Search from './pages/Search/Search';
import ResultSearch from './components/ResultSearch';
import { Provider } from './store';
import { EmptySearch } from './components/EmptySearch';
import ProductDetail from './pages/ProductDetail/ProductDetail';

export const App = () => {
	return (
		<Router>
			<Provider>
				<header className="row header">
					<Search />
				</header>
				<main className="container">
					<div className="row flex-column">
						<Route exact path="/" component={EmptySearch} />
						<Route exact path="/items/:id" component={ProductDetail} />
						<Route exact path="/items" component={ResultSearch} />
						<Route component={EmptySearch} />
					</div>
				</main>
			</Provider>
		</Router>
	);
};

export default App;
