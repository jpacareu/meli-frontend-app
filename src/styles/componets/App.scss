.header {
    background-color: $main;
}

body {
    background-color: $white-gray;
}

.spinner__container {
    text-align: center;
    margin: 50px auto;
    width: 100%;
}