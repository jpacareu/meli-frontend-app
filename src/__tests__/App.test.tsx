import React from 'react';
import App from '../App';
import renderer, { ReactTestRenderer } from 'react-test-renderer';

let component: JSX.Element | null = null;
let tree: ReactTestRenderer;

describe('<App />', () => {
	beforeEach(() => {
		component = <App />;
		tree = renderer.create(component);
	});
	it('should render without crashing', () => {
		expect(tree.toJSON()).toMatchSnapshot();
	});
});
