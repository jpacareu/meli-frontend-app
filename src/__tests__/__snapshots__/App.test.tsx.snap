// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render without crashing 1`] = `
Array [
  <header
    className="row header"
  >
    <div
      className="offset-md-1 col-md-1 col-2"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="Logo Marcado Libre"
          className="search-bar__image"
          src="Logo_ML.png"
        />
      </a>
    </div>
    <div
      className="col-md-9 col-10"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="search-bar__container"
        >
          <input
            className="search-bar__input"
            onChange={[Function]}
            placeholder="Nunca dejes de buscar"
            value=""
          />
          <button
            className="search-bar__button"
            onClick={[Function]}
            type="submit"
          >
            <img
              alt="Search"
              src="ic_Search.png"
            />
          </button>
        </div>
      </form>
    </div>
  </header>,
  <main
    className="container"
  >
    <div
      className="row flex-column"
    />
  </main>,
]
`;
